File: LC/Asteroids.cpp
#include <vector>
#include <stack>
#include <iostream>
using namespace std;

vector<int> asteroidCollision(vector<int>& asteroids) {
    vector<int> stk;
    for(auto i : asteroids) {
        if(i > 0 || stk.empty())
            stk.push_back(i);
        else {
            while(!stk.empty() && stk[stk.size()-1] > 0 && stk[stk.size()-1] < abs(i))
                stk.pop_back();
            if(!stk.empty() && stk[stk.size()-1] == abs(i))
                stk.pop_back();
            else if (stk.empty() || stk[stk.size()-1] < 0)
                stk.push_back(i);
        }
    }
    return stk;
}

int main() {
    vector<int> asteroids = {5, 10, -5};
    vector<int> ans = asteroidCollision(asteroids);
    for(auto i : ans)
        cout << i << " ";
    cout << endl;
    return 0;
}

File: LC/Asteroids.py
asteroids = list(map(int, input().split()))

stk = []
for i in asteroids:
    if i > 0 and stk:
        stk.append(i)
    else:
        while stk and stk[-1] > 0 and stk[-1] < abs(i):
            stk.pop()
        if stk and stk[-1] == -i:
            stk.pop()
        elif not stk or stk[-1] < 0:
            stk.append(i)
print(*stk)